rllib/env/env_runner_group.py

```diff
@@ -1046,13 +1027,12 @@ def foreach_env_runner_async(
             remote_actor_ids=remote_worker_ids,
         )
 
-    @DeveloperAPI
     def fetch_ready_async_reqs(
         self,
         *,
         timeout_seconds: Optional[float] = 0.0,
         return_obj_refs: bool = False,
-        mark_healthy: bool = True,
+        mark_healthy: bool = False,
```

Comment:
Most important change here:
It's possible that an async result is still in the object store as the actor has crashed and been restarted. Fetching a valid result from such a restarted actor should NOT by default mark it as healthy. Only dedicated health checks, such as `EnvRunnerGroup.probe_unhealthy_workers` should set `mark_healthy=True`.